lumen/sources/duckdb.py

```diff
@@ -134,6 +136,35 @@ def __init__(self, **params):
 
             self.tables = processed_tables
 
+        # Setup mirrors on first connection access
+        if self.mirrors:
+            _ = self._connection  # This will trigger connection creation and mirror setup
```

Comment:
Why even assign it to a variable? I think it's also cleaner to move the initialization logic into a method and call that here (and in the `_connection` property).